lualib/rspamadm/mime.lua

```diff
@@ -873,19 +879,85 @@ local function sign_handler(opts)
   end
 end
 
+-- Strips directories and .extensions (if present) from a filepath
+  -- very_simple
+  -- /home/very_simple.eml
+  -- very_simple.eml
+  -- very_simple.example.eml
+  -- /home/very_simple
+  -- home/very_simple
+  -- ./home/very_simple
+  -- ../home/very_simple.eml
+--All the above end up as very_simple
+local function filename_only(filepath)
+  filename = filepath:match(".*%/([^%.]+)")
+  if not filename then
+    filename = filepath:match("([^%.]+)")
+  end
+  return filename
+end
+
+--Write the dump content to file or standard out
+local function write_dump_content(dump_content, fname, extension, outdir)
+  
+  wrote_filepath = nil
+  if outdir then
+    if outdir:sub(-1) ~= "/" then
+      outdir = outdir .. "/"
+    end
+
+    local outpath = string.format("%s%s.%s", outdir, filename_only(fname), extension)
+    local outfile = io.open(outpath, "w")
+
+    if outfile then
+      outfile:write(dump_content)
+      outfile:close()
+      io.write(outpath.."\n")
+      wrote_filepath = outpath
+    else
+      io.stderr:write(string.format("Unable to open: %s\n", outpath))
+    end
+  else
+      io.write(dump_content)
+  end
+
+  return wrote_filepath
+end
+
+-- Get the formatted ucl (split or unsplit) or the raw task content
+local function get_dump_content(task, opts, fname)
+  if opts.ucl or opts.json or opts.messagepack then
+    local ucl_object = lua_mime.message_to_ucl(task)
+    
+    --Split out the content field into separate raws and update the ucl
+    if opts.split then
+      for i, part in ipairs(ucl_object.parts) do
+        if part.content then
+          local part_filename = string.format("%s-part%d", filename_only(fname), i)
+          local part_path = write_dump_content(tostring(part.content), part_filename, "raw", opts.outdir)
+          if part_path then
+            part.content = ucl.null
+            part.content_path = part_path
+          end
+        end
+      end
+    end
+
+    local extension = output_fmt(opts)
+    return ucl.to_format(ucl_object, extension), extension
+  end
+  return tostring(task:get_content()), "mime"
```

Comment:
Same here - it is not clear to me why `tostring` is used.

Reply:
as above